### DISCUSSION TITLE
Docker Discussion

### DISCUSSION THREAD

**Discussion by Hemanth0324** [DrAlzahraniProjects/csusb_fall2024_cse6550_team4#7] [Polls]
I request everyone to participate here related to Docker

**Comment by venkat-coyote** [comment]
Docker file is running successfully in my computer.
OS : Windows 11

![Screenshot 2024-09-12 205235](https://github.com/user-attachments/assets/cef27680-aa4d-4828-adef-ebcbb2412dd3)

**Comment by Jaya1809** [comment]
Docker file is running successfully in my computer
OS: Windows 11
![Screenshot (14)](https://github.com/user-attachments/assets/96dd7879-8b92-4922-a3fd-298842291792)

**Comment by Vaishnaviuppari37** [comment]
I have pushed the latest code of dockerfile into github which is running perfectly in my computer with necessary requirements.
OS: macOS

**Comment by venkat-coyote** [comment]
These are some basic docker commands which will be helpful have a look
### Docker Basics: A Quick Guide

**Docker** is an open platform designed to help developers build, ship, and run applications in a standardized way. It allows you to package your application, along with all its dependencies, into a single unit called a **container**. Containers are lightweight, portable, and ensure consistency across various environments, from development to production.

---

### Key Concepts

1. **Container**: A lightweight, standalone, and executable package that includes everything needed to run a piece of software, such as code, runtime, libraries, and settings.
2. **Image**: A read-only template used to create containers. Images are built from a set of instructions called a **Dockerfile**.
3. **Dockerfile**: A text file that contains a series of commands to assemble an image.
4. **Registry**: A storage and distribution system for Docker images, such as Docker Hub.
5. **Volume**: A mechanism for persisting data generated and used by Docker containers.

---

### Basic Docker Commands

1. **Check Docker Version**:
   ```bash
   docker --version
   ```
   Displays the installed version of Docker.

2. **Pull an Image from Docker Hub**:
   ```bash
   docker pull <image-name>
   ```
   Downloads an image from Docker Hub (e.g., `docker pull nginx`).

3. **List Docker Images**:
   ```bash
   docker images
   ```
   Displays all images downloaded to your local system.

4. **Run a Docker Container**:
   ```bash
   docker run -d --name <container-name> <image-name>
   ```
   Runs a container from an image. The `-d` flag runs the container in detached mode (in the background).

   Example:
   ```bash
   docker run -d --name mynginx nginx
   ```

5. **List Running Containers**:
   ```bash
   docker ps
   ```
   Displays all currently running containers.

6. **Stop a Docker Container**:
   ```bash
   docker stop <container-name>
   ```
   Stops a running container.

7. **Remove a Docker Container**:
   ```bash
   docker rm <container-name>
   ```
   Deletes a stopped container.

8. **Remove a Docker Image**:
   ```bash
   docker rmi <image-name>
   ```
   Deletes an image from your local system.

9. **Build a Docker Image from a Dockerfile**:
   ```bash
   docker build -t <image-name> .
   ```
   Builds an image from the Dockerfile in the current directory. The `-t` flag tags the image with a name.

10. **Inspect Docker Container Details**:
    ```bash
    docker inspect <container-name>
    ```
    Provides detailed information about a container, including its configuration and state.

11. **Attach to a Running Docker Container**:
    ```bash
    docker exec -it <container-name> /bin/bash
    ```
    Opens an interactive terminal inside a running container.

12. **Check Docker System Resources**:
    ```bash
    docker system df
    ```
    Displays disk usage information of Docker images, containers, and volumes.

---

### Useful Tip

- **Docker Hub**: Visit [Docker Hub](https://hub.docker.com/) to explore and download popular Docker images.

**Comment by Mukesh008759823** [comment]
Docker file is running successfully in my computer.
OS : Windows 11

![Screenshot 2024-09-13 100713](https://github.com/user-attachments/assets/3c7b5749-ac65-48bc-b3d4-939abc8ed626)
